### PR TITLE
ramips: switch default SDHC driver to upstream version

### PR DESCRIPTION
**Pull request by DragonBluep**

### CODE REVIEW

File: target/linux/ramips/patches-6.6/831-mmc-mtk-sd-initialize-pad-delay-and-drive-strength.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ 
++	/* Set pins drive strength */
++	writel(0x000d0044, host->base + MSDC_PAD_CTRL0);
++	writel(0x000e0044, host->base + MSDC_PAD_CTRL1);
++	writel(0x000e0044, host->base + MSDC_PAD_CTRL2);
++
++	/* Set pad delay */
++	sdr_clr_bits(host->base + MSDC_PAD_TUNE, 0xffdf1f1f);
++	sdr_set_bits(host->base + MSDC_PAD_TUNE, 0x84101010);
++	writel(0x10101010, host->base + MSDC_PAD_TUNE0);
```
Comment (neheb) [inline target/linux/ramips/patches-6.6/831-mmc-mtk-sd-initialize-pad-delay-and-drive-strength.patch:28]: isn't tune0 handled already?

Comment (DragonBluep) [reply target/linux/ramips/patches-6.6/831-mmc-mtk-sd-initialize-pad-delay-and-drive-strength.patch:28]: MT762x only operates on register `MSDC_PAD_TUNE`. ref:
https://github.com/torvalds/linux/blob/256abd8e550ce977b728be79a74e1729438b4948/drivers/mmc/host/mtk-sd.c#L539

In our local driver. This register name `MSDC_DAT_RDDLY0`.
https://github.com/openwrt/openwrt/blob/e7599942af8e14541166234349e9e49278455ef6/target/linux/ramips/files/drivers/mmc/host/mtk-mmc/sd.c#L2107

**Comment by mrkiko** [comment]
Nice work.

**Comment by DragonBluep** [comment]
@john-tho @PolynomialDivision Based on https://github.com/openwrt/openwrt/pull/12738#discussion_r1278314904, I made some improvements. After testing, now it works well on MT7620 and MT7628.

**Comment by fkroessbacher** [comment]
Hy out there,

i'm on an MikroTik RouterBOARD 750Gr3 (MediaTek MT7621 ver:1 eco:4, ramips/mt7621) and today i tried the new Release OpenWrt 23.05.4, but it's not working. So i Switched back to the only working solution OpenWrt 22.03.6.

Should the Fix be in the new Version, or missed i something?

cheers

**Comment by DragonBluep** [comment]
> Should the Fix be in the new Version, or missed i something?

@fkroessbacher Only in snapshot version.

**Comment by noda-odon** [comment]
> @fkroessbacher Only in snapshot version.

@DragonBluep , today, I downloaded snapshot Tue Aug 20 01:33:07 2024
and the issue is still there

**Comment by PaddyTheShamrock** [comment]
> > @fkroessbacher Only in snapshot version.
> 
> @DragonBluep , today, I downloaded snapshot Tue Aug 20 01:33:07 2024 and the issue is still there

Try:

`opkg remove kmod-sdhci-mt7620`

`opkg install kmod-mmc-mtk`

Then reboot the router.

**Comment by noda-odon** [comment]
@PaddyTheShamrock , It works!
Thanks

BTW: I have two mmc cards, one have the reported issue, while another works OK without replacing modules.

**Pull request merged**